webapp/app/js/filters.js

```diff
@@ -39,11 +39,25 @@ treeherder.filter('stripHtml', function() {
 
 treeherder.filter('linkifyBugs', function() {
     return function(input) {
-        var re = new RegExp('(?:Bug (\\d+))', 'ig');
-        var str = input || '';
-        return str.replace(re,
-            '<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=$1" target="_blank" title="Bug $1">$1</a>'
-        );
+        if (input) {
+            var re = /.*-- ([0-9]+)|.*bug-([0-9]+)|.*Bug ([0-9]+)/ig; //Matches 3 paterrns "<-- XXX" , "bug-XXX" and "Bug XXX"
+            var bug_matches = re.exec(input);
+            var pr_matches = /PR#([0-9]+)/i.exec(input);
+            if (pr_matches && bug_matches ) { //Check if there are any matches, if not return the 'input'
```

Comment:
We should handle pr_matches and bug_matches seperately. The way this clause is working is that we'll only replace one if we replace them both. Some messages will only reference bug #'s, not PR's.